codegen/rust/src/header.rs

```diff
@@ -71,7 +71,9 @@ impl<'a> HeaderGenerator<'a> {
                     // the p4 confused-endian data model.
                     let mut v = b.into_vec();
                     v.reverse();
-                    let b = BitVec::<u8, Msb0>::from_vec(v);
+                    let mut b = BitVec::<u8, Msb0>::from_vec(v);
+                    b.shift_left(#offset % 8);
```

Comment:
This is the primary change in this PR. If a field offset is not on a byte boundary, we need to account for that when setting the value from an input buffer.